airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/retriers/default_retrier.py

```diff
@@ -2,27 +2,105 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
-from typing import Optional, Union
+from typing import List, Optional, Union
 
 import requests
-from airbyte_cdk.sources.declarative.requesters.retriers.retrier import Retrier
+from airbyte_cdk.sources.declarative.requesters.retriers.backoff_strategies.exponential_backoff_strategy import ExponentialBackoffStrategy
+from airbyte_cdk.sources.declarative.requesters.retriers.backoff_strategy import BackoffStrategy
+from airbyte_cdk.sources.declarative.requesters.retriers.http_response_filter import HttpResponseFilter
+from airbyte_cdk.sources.declarative.requesters.retriers.retrier import (
+    NonRetriableResponseStatus,
+    ResponseStatus,
+    Retrier,
+    RetryResponseStatus,
+)
 
 
 class DefaultRetrier(Retrier):
-    def __init__(self, max_retries: Optional[int] = 5, retry_factor: float = 5):
+    """
+    Sample configs:
```

Comment:
It would be useful to add a description of what is the behavior of this default retrier. Would make autogenned docs useful in the future too.

Reply:
Specifically some points which would be useful to clarify: 
* By default, what filter and backoff strategy does it use? 
* How does this work when used with multiple filters? 
* How does it work when used with multiple backoff strategies?